apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.UnitTests/SessionPoolTests.cs

```diff
@@ -0,0 +1,443 @@
+//            TestLogger.Install();
+//            Logger.LogLevel = V1.Logging.LogLevel.Debug;
+        }
+
+        //Creates a client that will log created sessions in the _createdSession dictionary.
+        private Mock<SpannerClient> CreateMockClient(DatabaseName dbName = null)
+        {
+
+            var mockClient = new Mock<SpannerClient>();
+            if (dbName == null)
+            {
+                dbName = s_defaultName;
+            }
+            _createdSessions.TryAdd(mockClient, new List<Session>());
+
+            mockClient.Setup(
+                    x => x.CreateSessionAsync(
+                        It.Is<DatabaseName>(y => y.Equals(dbName)),
+                        It.IsAny<CancellationToken>()))
+                .ReturnsAsync(
+                    () =>
+                    {
+                        var mockSession = new Session
+                        {
+                            Name = $"{s_defaultName}/sessions/{Guid.NewGuid()}"
+                        };
+                        List<Session> sessionList;
+                        if (_createdSessions.TryGetValue(mockClient, out sessionList))
+                        {
+                            lock (sessionList)
+                            {
+                                sessionList.Add(mockSession);
+                            }
+                        }
+                        return mockSession;
+                    });
+
+            mockClient.Setup(
+                    x => x.BeginTransactionAsync(It.IsAny<BeginTransactionRequest>(), It.IsAny<CallSettings>()))
+                .ReturnsAsync(
+                    () =>
+                    {
+                        var tx = new Transaction { Id = ByteString.CopyFromUtf8(Guid.NewGuid().ToString())};
+                        _transactions.Push(tx);
+                        return tx;
+                    });
+
+            mockClient.Setup(x => x.CommitAsync(It.IsAny<SessionName>(), It.IsAny<ByteString>(),
+                It.IsAny<IEnumerable<Mutation>>(), It.IsAny<CallSettings>()))
+                .ReturnsAsync(() => new CommitResponse());
+
+            return mockClient;
+        }
+
+        [Fact]
+        public void CanCreateSessionPool()
+        {
+            SessionPool pool = new SessionPool();
+            pool.Dispose();
+        }
+
+        [Fact]
+
+        public async Task EmptyPoolCreatesNewSession()
+        {
+            using (var pool = new SessionPool())
+            {
+
+                var client = CreateMockClient();
+                var session = await pool.CreateSessionFromPoolAsync(
+                        client.Object, s_defaultName.ProjectId,
+                        s_defaultName.InstanceId, s_defaultName.DatabaseId, null, CancellationToken.None)
+                    .ConfigureAwait(false);
+
+                Assert.Equal(1, _createdSessions.Count);
+                Assert.Equal(1, _createdSessions[client].Count);
```

Comment:
yes, because the test is over at this point.  xunit creates an instance of the test class per test method.